Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -97,4 +101,48 @@ class PurchasedProductsFetcher: PurchasedProductsFetcherType {
         return result
     }
 
+    private static var hasPendingConsumablePurchase: Bool {
```

Comment:
Is it possible to unit test these? I guess we would have to mock `StoreKit`. We do have integration tests so it's probably okish, but I think it's good to test things at different granularities to try to have as much coverage as possible.

Reply:
It's _indirectly_ tested, but I agree it would be useful to test independently 👍🏻  which we can do with `SKTestSession`.

I added a type in #2533 to do something similar, so I'm going to combine both things and test them.

Reply:
Extracted and tested in #2539 like you suggested.